src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using sfh = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	/// <summary>
+	/// 
+	/// </summary>
+	public abstract class HealthCheckPublisher : IHealthCheckPublisher
+	{
+		/// <summary>
+		/// 
+		/// </summary>
+		protected StringBuilder m_descriptionBuilder;
+
+		/// <summary>
+		/// 
+		/// </summary>
+		protected string HealthReportSourceId { get { return HealthReportSourceIdImpl; } }
+
+		/// <summary>
+		/// 
+		/// </summary>
+		protected abstract string HealthReportSourceIdImpl { get; }
+
+		/// <summary>
+		/// 
+		/// </summary>
+		protected const string HealthReportSummaryProperty = "HealthReportSummary";
+
+		/// <summary>
+		/// 
+		/// </summary>
+		public HealthCheckPublisher()
+		{
+			m_descriptionBuilder = new();
```

Comment:
We should not create new instances of `StringBuilder` to avoid allocations. Looks how it was implemented before with `StringBuilderPool`.